### PR TITLE
Add Run AQA (Comment-Triggered PR Builds) workflow

### PR DESCRIPTION
**Pull request by Icohedron**
As per AdoptOpenJDK/openjdk-tests#2399, adds the Run AQA (Comment-Triggered PR Builds) workflow to the TKG repo.
When ran on a PR in this (TKG) repo, the [AdoptOpenJDK/run-aqa@v1](https://github.com/AdoptOpenJDK/run-aqa) GitHub action is ran with the `tkg_Repo` argument set to the head of the PR.

### CODE REVIEW

**Comment by lumpfish** [comment]
I've never written a github action, so perhaps not the best person to review this - but how can I invoke the action to test that it works?

**Comment by smlambert** [comment]
Not sure why @karianna has assigned @lumpfish and @Mesbah-Alam to review this (perhaps thought this was the STF repo).  I will remove you from reviewers list.

This is part of the CanOSP work, and a port of what is already delivered to the openjdk-tests repo (and documented here: https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/doc/RunAqa.md)

@Icohedron has run a test in his branch, https://github.com/Icohedron/TKG/pull/1

**Comment by karianna** [comment]
> Not sure why @karianna has assigned @lumpfish and @Mesbah-Alam to review this (perhaps thought this was the STF repo). I will remove you from reviewers list.
> 
> This is part of the CanOSP work, and a port of what is already delivered to the openjdk-tests repo (and documented here: https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/doc/RunAqa.md)
> 
> @Icohedron has run a test in his branch, [Icohedron#1](https://github.com/Icohedron/TKG/pull/1)

Yep in my tired state I read STF :-/

File: .github/workflows/runAqaArgParse.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    # Strip leading and trailing whitespace. Remove empty arguments that may result after stripping.
+    raw_args = list(filter(lambda empty: empty, map(lambda s: s.strip(), raw_args)))
+
+    parser = argparse.ArgumentParser(prog=keyword, add_help=False)
+    # Improvement: Automatically resolve the valid choices for each argument populate them below, rather than hard-coding choices.
+    parser.add_argument('--sdk_resource', default=['nightly'], choices=['nightly', 'releases', 'customized'], nargs='+')
+    parser.add_argument('--customized_sdk_url', default=['None'], nargs='+')
+    parser.add_argument('--archive_extension', default=['.tar'], choices=['.zip', '.tar', '.7z'], nargs='+')
+    parser.add_argument('--build_list', default=['openjdk'], choices=['openjdk', 'functional', 'system', 'perf', 'external'], nargs='+')
```
Comment (smlambert) [inline .github/workflows/runAqaArgParse.py:51]: I saw this in openjdk-tests repo, but not tried yet to find out what happens when someone tries to build external?  Is docker installed on github runners and this works smoothly?

Comment (sophia-guo) [reply .github/workflows/runAqaArgParse.py:51]: I've tried with jenkin_tests and camel tests and hit the same issue:
https://github.com/sophia-guo/runaqaTest/runs/2249338597?check_suite_focus=true
https://github.com/sophia-guo/runaqaTest/runs/2249127830?check_suite_focus=true

Look docker is fine. There are some hidden issues related with environment setup? Both jobs can successfully create docker images but some how `unable to recognize DOCKER_OS` hence docker run is trying to delete default images and find the image is not there. I will take a look.

```
Unable to recognize DOCKER_OS from DOCKERIMAGE_TAG = --version!
docker run  --rm adoptopenjdk-derby-test:8-jdk-ubuntu-hotspot-full 
Unable to find image 'adoptopenjdk-derby-test:8-jdk-ubuntu-hotspot-full' locally

```

**Review by renfeiw — APPROVED** [review]
The porting change looks good to me. For future improvement, I think it may be better to keep runAqaArgParse.py in a common place, for example under TKG/scripts/testBot, so we don't need to duplicate it in multiple repos.

**Comment by Icohedron** [comment]
> The porting change looks good to me. For future improvement, I think it may be better to keep runAqaArgParse.py in a common place, for example under TKG/scripts/testBot, so we don't need to duplicate it in multiple repos.

The defaults and/or available parameters may differ for each repo. 
For this TKG repo version of the workflow, the `tkg_repo` argument is not available because the TKG repo will always be the head of the PR. In the openjdk-test repo version of this workflow, a custom `tkg_repo` is able to be specified.
Hence, the `runAqaArgParse.py` is slightly different between the two versions of the workflow for the two repos.

**Review by smlambert — APPROVED** [review]
Indeed, in future if we put the script in a common place, it could be refactored to read defaults from a config file for each repo its contained within, but certainly can come later.

Thanks @Icohedron !

**Pull request merged**